docs/topics/mpp/mpp-publish-lib.md

```diff
@@ -122,40 +122,14 @@ kotlin {
 The example works for Android libraries without [product flavors](https://developer.android.com/studio/build/build-variants#product-flavors). 
 For a library with product flavors, the variant names also contain the flavors, like `fooBarDebug` or `fooBazRelease`.
 
-> If a library consumer defines variants that are missing in the library, they need to provide matching fallbacks. 
-> For example, if a library does not have or does not publish a staging build type, the library consumer must provide a fallback for the 
-> consumers who have such a build type, specifying at least one of the build types that the library publishes:
->
-><tabs>
->
-> ```groovy
-> android {
->     buildTypes {
->         staging {
->             // ...
->             matchingFallbacks = ['release', 'debug']
->         }
->     }
-> }
-> ```
-> 
-> ```kotlin
-> android {
->     buildTypes {
->         val staging by creating {
->             // ...
->             matchingFallbacks = listOf("release", "debug")
->         }
->     }
-> }
-> ```
->
-></tabs>
->
-{type="note"}
-
-Similarly, a library consumer needs to provide matching fallbacks for custom product flavors if some are missing in the 
-library publications.
+The default publishing setup is the following:
+* If all Android variants that the project publishes have the same build type attribute, then the published variants 
```

Comment:
I'm not sure that mentioning the attribute values here makes things clearer, as attributes are a lower-level detail of dependency resolution than an ordinary build author usually cares about.

Maybe we can somehow reformulate this. Instead of describing the attribute values we can tell about compatibility rules, like:

> If all published Android variants have the same build type (`debug`, `release` etc.), they will be compatible with any consumer's build type.
> If the published variants have different build types, then the `release` variants will also be compatible with any consumer's build type that is absent among the published variants, while all other variants (like `debug`) will only match the same build type on the consumer side, unless the consumer project specifies the matching fallbacks. 

So the part about matching fallbacks might still be relevant for the second case. However, matching fallbacks are no longer necessary specifically for MPP, so maybe a link to the AGP docs would be enough.

Reply:
Thank you! Just a little bit modified your option and changed the wording in the last sentence on how to opt out. Please take a look.

Reply:
Looks good to me. 👍